packages/xo-web/src/xo-app/host/tab-stats.js

```diff
@@ -32,6 +38,31 @@ export default class HostStats extends Component {
       if (cancelled) {
         return
       }
+      const statsPifs = stats.stats.pifs
+      const { host, pifs } = this.props
+
+      const deviceName = []
+      forEach(host.PIFs, currentPif => {
+        if (pifs[currentPif]) {
+          if (!deviceName.includes(pifs[currentPif].device)) {
+            deviceName.push(pifs[currentPif].device)
+          }
+        }
+      })
+
+      for (const [signal, pifs] of Object.entries(statsPifs)) {
+        forEach(pifs, (pif, i) => {
+          if (signal === 'rx') {
+            delete statsPifs.rx[i]
+            statsPifs.rx[deviceName[i]] = pif
```

Comment:
There is no guarantee that you will get the right device name with the same index.